### DISCUSSION TITLE
Sort out org and repo admins

### DISCUSSION THREAD

**Discussion by yaxu** [tidalcycles/Tidal#1125] [General]
I've been quite liberal and disorganised in giving people privileges to the tidal repository over the years. I'm not sure if anyone's ever asked for access, I've just given it out so that people can do stuff. It'd be good to tidy this up a bit.

The current picture:

Org admins:
* @bgold-cosmos
* @dktr0 
* @jarmitage 
* @kindohm 
* @lwlsn 
* @munshkr
* @ndr-brt 
* @sfradkin 
* @telephon 
* @yaxu 

Org 'members':
* @calumgunn 
* @lennart 
* @msp 
* @nini-faroux 
* @polymorphicengine 
* @willbasky 

Repo collaborators:
* @Bubobubobubobubo (tidal-doc)
* @bwagner (tidalink)
* @cleary (clean-samples)
* @dowdiness (tidal-doc)
* @TylerMclaughlin (tidal)

There are also people contributing PRs without privileges, like @mindofmatthew.

I guess there are different reasons to give access:
* To allow active people to approve PRs etc
* To have some trusted people who can take care of things (i.e. give others admin rights) if others drop out of action for any reason
* To encourage contributions
* ?

So.. Who (whether in the above list or not) would be happy to keep or gain this access? Who wants to be involved with actively doing admin stuff like approving PRs etc, and on what repositories? How would you like this role described?

All thoughts very welcome!

**Comment by cleary** [comment]
I guess what's not clear to me is how much repository maintenance goes on across the tidalcycles repos - is there specific stuff that is menial and time consuming that would be good for interested parties such as myself to spend some time on (eg bug triage, documentation)? 

I'm so green to haskell so I don't have any particular interest in having commit/pr approve access, but I'd be happy to do a bit of maintenance to familiarise myself with the processes

**Comment by kindohm** [comment]
I am happy to help triage/verify bugs, help respond to issues, test PR's and fixes, and help guide conversations. My involvement in this repo has been minimal as of late, but I would be happy to help in these areas. 

While I currently have permission to merge PRs, I don't think I would ever exercise that privilege. I just don't know Haskell well enough to recognize a glaring issue in a contribution. 

Since my contribution ends at testing PRs and being involved in issue discussion, I'm not sure it makes sense for me to remain an admin. I should probably be downgraded to "member".

**Comment by ndr-brt** [comment]
I'd like to keep the admin role, at least on the atom plugin repo.
On the main repo I like to contribute, responding to the issues, opening/reviewing/approving PRs, I never allowed myself to merge a PR because I always thought that the decision to merge a PR should be up to you and I think this should be a good practice to keep (your time permitting).

**Comment by jarmitage** [comment]
> To have some trusted people who can take care of things (i.e. give others admin rights) if others drop out of action for any reason
> To encourage contributions

I'm happy to keep my role for these reasons

**Comment by Bubobubobubobubo** [comment]
I would like to keep the collaborator role on tidal-docs and thanks again for your trust. I have planned to contribute to the docs again whenever I will have the time and the energy to do so!

**Comment by yaxu** [comment]
Hm well we have a lot of open issues, it would be good to come up with a triage crew to tidy them up a bit. For example I've seen some repos put tag more speculative issues with 'wishlist' and then close them. I think there is not much menial work involved to share around @cleary, but still it would be good to share the responsibility so we get more organised and more responsive, and therefore encourage more contributions.

I'm happy to continue as PR decision-maker for consistency, I hope to have more time for such stuff from December.. But we could also move towards some system where we require X number of approvals to do a merge.

**Comment by polymorphicengine** [comment]
I'd like to get more active in contributing and responding / looking at issues. I think my current position as member is ok for that though. Maybe in the future, when we possibly also move the tidal-gui from my repo I could take more responsibility.

**Comment by bgold-cosmos** [comment]
I unfortunately haven't been very active over the past year or so but I think I've used the admin privs in the past to merge PRs, so it's been nice to have that option.

If issue cleanup is a problem, then maybe some tags where people could "claim" (maybe nonexclusively, though?) an issue might help?  I think sometimes I'm prone to leaving things alone if I think it's in somebody else's domain, but putting my name on something would motivate me to make sure something's getting done.  I don't mind being nominated for stuff, either, but it can be hard for me to promise regular progress...

**Comment by dktr0** [comment]
No strong opinions here one way or the other. Happy to stay where I am or be moved, whatever works. I've usually been leaving the merging of PRs to @yaxu anyway, or after his review, although maybe there is the odd time where I just do it when I can see it's trivial.

**Comment by willbasky** [comment]
Hi, Alex.
I updated and refactored `tidal-vis`. After that i dreamed to contribute to tidal itself, it not came true unfortunately. I'd like to be in touch, not active for now. Up to you.

**Comment by yaxu** [comment]
I should say that there is/was no thinking between 'admins', 'members' and 'collaborators', I've fairly randomly assigned things according to whatever setting I found in github's interface first..

There's also little difference as (AFAIK) they can all merge pull requests, do releases and so on. I haven't checked but I think 'collaborators' just have access to particular repositories not all of them, and 'admins' have the ability to change other people's permissions.

So one idea would be to make around three people admins, which could be the same three that administrate of the opencollective fund. Then make anyone else who wants it/would find it useful team 'members' as a fairly flat structure.

**Comment by lwlsn** [comment]
Heya, happy to stay as actively involved as possible. Same as @ndr-brt et al that I haven't felt comfortable merging any PRs with things like specific features as figured it should go through @yaxu. But happy to take on more of an active role with maintaining any other aspects. The admin/team member structure sounds good though!

**Comment by munshkr** [comment]
I was originally given admin priviledge to migrate the vim-tidal repository to the organization, so I'm fine with being just an org member and/or a repo collaborator for that repository only, whatever option is the easiest/makes sense.

**Comment by msp** [comment]
Hi all

I made some 1.0 commits as an effort to make a small contribution to the
community and try to learn some more Haskell.

I'd really love to do more of that but realistically will struggle to make
it a priority anytime soon. I wouldn't merge a PR unless it was trivial or
an area I understood well.

I'd like to remain part of the organisation but understand if you prefer to
only keep more active members onboard.

Alex - I think in Github terms, a collaborator is someone with commit privs
that isn't part of Tidal organisation.

Cheers

--
*Matt Spendlove*
+44 (0)7976 609379

________________________________________________
Cenatus <http://cenatus.org/> :: Waveform Transmissions
<http://spatial.infrasonics.net/> :: 00101010 <https://github.com/msp>
